applications/tasks/src/modules/kt_fax_cleanup.erl

```diff
@@ -0,0 +1,306 @@
+                      ,Doc
+                      ).
+
+%%------------------------------------------------------------------------------
+%% @doc util functions
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec format_modb(kzd_fax:doc()) ->
+                         {kz_term:ne_binary(), kz_term:ne_binary(), kz_term:ne_binary(), kz_term:ne_binary()}.
+format_modb(Doc) ->
+    FromId = kz_doc:id(Doc),
+    FromDB = kz_doc:account_db(Doc),
+    AccountId = kz_doc:account_id(Doc),
+    {Year, Month, _D} = kz_term:to_date(kz_doc:created(Doc)),
+    AccountMODb = kazoo_modb:get_modb(AccountId, Year, Month),
+    ToDB = kz_util:format_account_modb(AccountMODb, 'encoded'),
+    ToId = ?MATCH_MODB_PREFIX(kz_term:to_binary(Year), kz_date:pad_month(Month), FromId),
+    {FromDB, FromId, ToDB, ToId}.
+
+-spec is_stale_doc(kzd_fax:doc()) -> boolean().
+is_stale_doc(Doc) ->
+    (kz_time:now_s() - ?DEFAULT_AGE_OUT) >= kz_doc:created(Doc).
+
+-spec bump(map(), kz_term:ne_binary()) -> map().
+bump(State, Key) ->
+    bump(State, Key, 1).
+
+-spec bump(map(), kz_term:ne_binary(), integer()) -> map().
+bump(#{<<"counter">> := Counter}=State, Key, Value) ->
+    State#{<<"counter">> => Counter#{Key => (maps:get(Key, Counter, 0) + Value)}}.
+
+-spec has_attachments(kzd_fax:doc()) -> boolean().
+has_attachments(Doc) ->
+    case kz_doc:attachments(Doc) of
+        'undefined' -> 'false';
+        _Att -> 'true'
+    end.
+
+-spec list_attachments(kzd_fax:doc()) -> kz_term:ne_binaries().
+list_attachments(Doc) ->
+    case kz_doc:attachments(Doc) of
+        'undefined' -> [];
+        Att -> kz_json:get_keys(Att)
+    end.
+
+-spec maybe_fetch_account(map(), kz_term:ne_binary()) -> map().
+maybe_fetch_account(State, 'undefined') -> State;
+maybe_fetch_account(#{<<"accounts">> := Accounts}=State, AccountId) ->
+    case maps:get(AccountId, Accounts, 'undefined') of
+        'undefined' ->
+            fetch_account(State, AccountId);
+        _Found ->
+            State
+    end.
+
+-spec fetch_account(map(), kz_term:ne_binary()) -> map().
+fetch_account(#{<<"accounts">> := Accounts}=State, AccountId) ->
+    case kzd_accounts:fetch(AccountId) of
+        {'ok', _} ->
+            State#{<<"accounts">> => Accounts#{AccountId => 'true'}};
+        _Else ->
+            State#{<<"accounts">> => Accounts#{AccountId => 'false'}}
+    end.
+
+-spec account_exists(map(), kz_term:ne_binary()) -> boolean().
+account_exists(_State, 'undefined') -> 'false';
+account_exists(#{<<"accounts">> := Accounts}, AccountId) ->
+    maps:get(AccountId, Accounts, 'false').
+
+-spec retries_exhausted(kzd_fax:doc()) -> boolean().
+retries_exhausted(Doc) ->
+    kzd_fax:retries(Doc, 0) =:= kzd_fax:attempts(Doc, 0).
+
+-spec next_start_key(kz_json:objects()) -> {kz_term:ne_binary(), kz_json:objects()}.
+next_start_key([]) -> [];
+next_start_key(Docs) ->
+    {Results, [StartDoc]} = lists:split(length(Docs) -1, Docs),
```

Comment:
I wonder what the execution comparison of this compared to:

```erlang
[StartDoc | RevResults] = lists:reverse(Docs),
{kz_doc:id(StartDoc), lists:reverse(RevResults)}.
```

What might be a "worst-case" length for `Docs`?

Reply:
I was curious so I tested.

With lists of 10,000 elements, the average diff is 100 microseconds between the two approaches. Ignore for now :)